x/trigger/simulation/genesis.go

```diff
@@ -0,0 +1,140 @@
+package simulation
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/provenance-io/provenance/internal/pioconfig"
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+const (
+	TriggerID  = "trigger_id"
+	QueueStart = "queue_start"
+)
+
+// TriggerIDFn randomized MinActions
+func TriggerIDFn(r *rand.Rand) uint64 {
+	return uint64(simtypes.RandIntBetween(r, 2, 10000000000))
+}
+
+// QueueStartFn randomized MinActions
+func QueueStartFn(r *rand.Rand) uint64 {
+	return uint64(simtypes.RandIntBetween(r, 0, 10000000000))
+}
+
+// GetRandomEvent returns a random event
+func GetRandomEvent(r *rand.Rand, now time.Time) types.TriggerEventI {
+	rand := simtypes.RandIntBetween(r, 0, 1)
+	if rand > 0 {
+		randTime := now.Add(time.Hour * time.Duration(simtypes.RandIntBetween(r, 1, 10)))
+		return &types.BlockTimeEvent{Time: randTime.UTC()}
+	}
+	height := uint64(simtypes.RandIntBetween(r, 100000, 200000))
+	return &types.BlockHeightEvent{BlockHeight: height}
+}
+
+// GetRandomAction returns a random action
+func GetRandomAction(r *rand.Rand, from string, to string) sdk.Msg {
+	amount := int64(simtypes.RandIntBetween(r, 100, 1000))
+	return &banktypes.MsgSend{
+		FromAddress: from,
+		ToAddress:   to,
+		Amount:      sdk.NewCoins(sdk.NewInt64Coin(pioconfig.GetProvenanceConfig().BondDenom, amount)),
+	}
+}
+
+// GetRandomTrigger returns a random trigger
+func GetRandomTrigger(r *rand.Rand, simState *module.SimulationState) types.Trigger {
+	account := int64(simtypes.RandIntBetween(r, 1, 2))
```

Comment:
1. `RandIntBetween(r, 1, 2)` will always return `1`.
2. Probably better to use `len(simState.Accounts)+1` for the max.
3. Need to take into account the scenario where there's only 1 account. We don't want a panic in that case, which we'll get as-is from `simState.Accounts[account]`.

Even better, provide the accounts to `GetRandomAction` and have it pick the two to use. To make sure they're not the same, the best way is to randomize the slice (see `r.Shuffle`) and then take the first two.

It's pretty common to have an `accs []simtypes.Account` parameter in the sim functions. I'm pretty sure you could just provide `simState.Accounts` for such an argument. There's also a `simtypes.RandomAcc(r *rand.Rand, accs []Account) (Account, int)` function that might be useful. But be careful using it to get two different accounts. Make sure you don't end up with an infinite loop (e.g. when there's only 1 account and you're trying to get two different ones); that's why I suggested shuffling the list and taking the first two.